protocols/gossipsub/tests/smoke.rs

```diff
@@ -0,0 +1,226 @@
+                        Async::Ready((_addr, ev)) => {
+                            if f(ev) {
+                                return Ok(Async::Ready(this.take().unwrap()));
+                            }
+                        }
+                        Async::NotReady => return Ok(Async::NotReady),
+                    }
+                },
+                None => panic!("future called after final return"),
+            }
+        })
+        .timeout(Duration::from_secs(10))
+        .map_err(|e| panic!("{:?}", e));
+
+        rt.block_on(fut).unwrap()
+    }
+}
+
+fn build_node() -> (Multiaddr, TestSwarm) {
+    let key = identity::Keypair::generate_ed25519();
+    let public_key = key.public();
+
+    let transport = MemoryTransport::default()
+        .upgrade(upgrade::Version::V1)
+        .authenticate(PlainText2Config {
+            local_public_key: public_key.clone(),
+        })
+        .multiplex(yamux::Config::default())
+        .map(|(p, m), _| (p, StreamMuxerBox::new(m)))
+        .map_err(|e| panic!("Failed to create transport: {:?}", e))
+        .boxed();
+
+    let peer_id = public_key.clone().into_peer_id();
+    let behaviour = Gossipsub::new(peer_id.clone(), GossipsubConfig::default());
+    let mut swarm = Swarm::new(transport, behaviour, peer_id);
+
+    let port = 1 + random::<u64>();
+    let mut addr: Multiaddr = Protocol::Memory(port).into();
+    Swarm::listen_on(&mut swarm, addr.clone()).unwrap();
+
+    addr = addr.with(libp2p::core::multiaddr::Protocol::P2p(
+        public_key.into_peer_id().into(),
+    ));
+
+    (addr, swarm)
+}
+
+#[test]
+fn multi_hop_propagation() {
+    let _ = env_logger::try_init();
+
+    fn prop(num_nodes: usize, seed: u64) -> TestResult {
+        if num_nodes < 2 || num_nodes > 100 {
+            return TestResult::discard();
+        }
+
+        debug!("number nodes: {:?}, seed: {:?}", num_nodes, seed);
+
+        let mut rt = Runtime::new().unwrap();
+
+        let mut graph = Graph::new_connected(num_nodes, seed);
+        let number_nodes = graph.nodes.len();
+
+        // Subscribe each node to the same topic.
+        let topic = Topic::new("test-net".into());
+        for (_addr, node) in &mut graph.nodes {
+            node.subscribe(topic.clone());
+        }
+
+        // Wait for all nodes to be subscribed.
+        let mut subscribed = 0;
+        graph = graph.wait_for(&mut rt, move |ev| {
+            if let GossipsubEvent::Subscribed { .. } = ev {
+                subscribed += 1;
+                if subscribed == (number_nodes - 1) * 2 {
+                    return true;
+                }
+            }
+
+            false
+        });
+
+        // Publish a single message.
+        graph.nodes[0].1.publish(&topic, vec![1, 2, 3]);
+
+        // Wait for all nodes to receive the published message.
+        let mut received_msgs = 0;
+        graph.wait_for(&mut rt, move |ev| {
+            if let GossipsubEvent::Message(_peer_id, _message_id, msg) = ev {
```

Comment:
```suggestion
            if let GossipsubEvent::Message(_peer_id, _message_id, _msg) = ev {
```
To suppress the unused variable warning